SU2_CFD/src/output/COutput.cpp

```diff
@@ -347,11 +350,28 @@ void COutput::WriteToFile(CConfig *config, CGeometry *geometry, unsigned short f
 
     case RESTART_BINARY:
 
-      if (fileName.empty())
+      if (fileName.empty()){
         fileName = config->GetFilename(restartFilename, "", curTimeIter);
 
+        if (!config->GetWrt_Restart_Overwrite()){
+          if (config->GetMultizone_Problem()){
+            /* --- do nothing --- */  
+          }
+          else {
+
+            filename_iter = fileName;
+            filename_iter.append(inner_iter_ss.str());
+            filename_iter.append(CSU2BinaryFileWriter::fileExt);
+
+          }
+        }
+      }
+      
       if (rank == MASTER_NODE) {
-          (*fileWritingTable) << "SU2 restart" << fileName + CSU2BinaryFileWriter::fileExt;
+        if (!config->GetWrt_Restart_Overwrite())
+          (*fileWritingTable) << "SU2 restart with iteration" << filename_iter ;
+
+        (*fileWritingTable) << "SU2 restart" << fileName + CSU2BinaryFileWriter::fileExt;
```

Comment:
Before you apply this to the other types, put this block in a private function, to which you pass part of the description ("SU2 restart") and the file extension.